exec/java-exec/src/main/java/org/apache/drill/exec/store/RecordDataType.java

```diff
@@ -47,17 +48,22 @@
    * @return the constructed type
    */
   public final RelDataType getRowType(RelDataTypeFactory factory) {
-    final List<SqlTypeName> types = getFieldSqlTypeNames();
+    final List<ImmutablePair<SqlTypeName, Boolean>> types = getFieldSqlTypeNames();
```

Comment:
Better, but we still have three correlated lists. Can you combine the four fields into a single struct (class) and have a single list?

Reply:
There are just 2 lists : `names` and `types`. 
The third list (`fields`) is used as `factory.createStructType(fields, names)`
I'm reluctant to change abstract methods because they might have usage beyond the current SystemTable context.

Reply:
I also agree with Paul, using Pair is not a good choice. Better to introduce holder class.

Reply:
For a one-time usage of creating a nullable RelDataType (since Calcite doesn't provide a convenient way of doing so), I think a holder class is an overkill. 
I have this option from Calcite: 
`https://calcite.apache.org/apidocs/org/apache/calcite/rel/type/RelDataTypeFactory.html#createStructType-java.util.List-`
which also makes use of a list of Map.Entry objects. 
So, unless I change Calcite, there is no real value in creating a holder class.

Reply:
Ok, don't have strong opinion about this right now, so you may leave as is. But I would prefer Java native implementation in this case:
```
      AbstractMap.SimpleImmutableEntry<Integer, String> entry
          = new AbstractMap.SimpleImmutableEntry<>(1, "one");
```